pkg/kernel/Dockerfile

```diff
@@ -137,6 +137,18 @@ RUN make CROSS_COMPILE="${CROSS_COMPILE_ENV}" ARCH="${KERNEL_ARCH}" oldconfig
 # diffconfig returns 0 in all cases, grep to throw error in a case config has changed between build stages.
 RUN ! scripts/diffconfig .config .config.new | grep .
 
+# To have a reproducible kernel build
+ARG KBUILD_BUILD_TIMESTAMP
+ARG KBUILD_BUILD_USER
+ARG KBUILD_BUILD_HOST
+ARG KCONFIG_NOTIMESTAMP
+ARG SOURCE_DATE_EPOCH
+ENV KBUILD_BUILD_TIMESTAMP=${KBUILD_BUILD_TIMESTAMP}
+ENV KBUILD_BUILD_USER=${KBUILD_BUILD_USER}
+ENV KBUILD_BUILD_HOST=${KBUILD_BUILD_HOST}
+ENV KCONFIG_NOTIMESTAMP=${KCONFIG_NOTIMESTAMP}
+ENV SOURCE_DATE_EPOCH=${SOURCE_DATE_EPOCH}
```

Comment:
Is it useful to record these somewhere? Or do they all get recorded in the version string for the built kernel?

Reply:
they get recorded : 
```
linuxkit-525400123456:~# uname -a
Linux linuxkit-525400123456 5.10.121-default #1 SMP Tue Apr 18 14:25:31 2023 +0000 aarch64 Linux
linuxkit-525400123456:~# cat /proc/version
Linux version 5.10.121-default (eve@eve) (gcc (Alpine 11.2.1_git20220219) 11.2.1 20220219, GNU ld (GNU Binutils) 2.38) #1 SMP Tue Apr 18 14:25:31 2023 +0000
```